core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -528,4 +544,21 @@ private class AppStatusListener(kvstore: KVStore) extends SparkListener with Log
     entity.write(kvstore)
   }
 
+  /** Update a live entity only if it hasn't been updated in the last configured period. */
+  private def maybeUpdate(entity: LiveEntity): Unit = {
+    if (liveUpdatePeriodNs >= 0) {
+      val now = System.nanoTime()
```

Comment:
I didn't notice that method ever showing up when profiling; my guess is it's just a read from some CPU register (TSC?) and so reasonably cheap.

Reply:
I think it can vary a lot with OS (and maybe the hardware?)  Unfortunately when searching now, most of the references are really dated, I have no idea what info is obsolete.  But there is enough evidence it seems prudent to avoid calling it a lot in case its slow in some situations.